### PR TITLE
Development

### PR DESCRIPTION
**Pull request by 1ba1**

### CODE REVIEW

File: assets/js/scripts.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+function Book(title, author, pages, read) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.read = read;
+}
+
+Book.prototype.toggle = function() {
+  this.read ? (this.read = false) : (this.read = true);
```
Comment (rvvergara) [inline assets/js/scripts.js]: We can shorten this to:

```javascript
this.read = !this.read
```

File: assets/js/scripts.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    read.textContent = book.read ? "Read" : "Not read yet";
+    read.onclick = () => {
+      book.toggle();
+      localStorage.setItem("book-collection", JSON.stringify(myLibrary));
+      render();
+    };
+    remove.innerHTML = "×";
+    remove.addEventListener("click", removeBookFromLibrary, false);
+
+    card.appendChild(header);
```
Comment (rvvergara) [inline assets/js/scripts.js]: We can write a separate function that simply will take in `card` and whatever you wish to append to it:

```javascript
// this function should be declared outside of render
const appendToParent = (parent, child) => parent.appendChild(child)
```
To add all the children elements:

```javascript
[header, title, author, pages, read, remove].forEach(child => appendToParent(card, child);

card.setAttribute('data-index', index);

appendToParent(container, card);
```

File: assets/js/scripts.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+function render() {
+  const container = document.getElementsByClassName("container")[0];
+  while (container.hasChildNodes()) {
+    container.removeChild(container.firstChild);
+  }
+  myLibrary.forEach((book, index) => {
+    book.__proto__ = Object.create(new Book);
+    const card = document.createElement("div"),
+      header = document.createElement("header"),
```
Comment (rvvergara) [inline assets/js/scripts.js]: It's better we do separate `const` statements for each variable we are declaring. I am using the AirBnB styleguide as a standard (at least in my case I normally have ESlint catch them for me).

File: assets/js/scripts.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      title = document.createElement("p"),
+      author = document.createElement("p"),
+      pages = document.createElement("p"),
+      read = document.createElement("button"),
+      remove = document.createElement("div");
+
+    card.setAttribute("class", "card");
```
Comment (rvvergara) [inline assets/js/scripts.js]: Like I suggested below (`appendToParent` function) we can dry this up by having a separate function that sets the attributes. Then we can simply use an array to store all the different inputs we'll need (as I showed in the `appendToParent`).

File: assets/js/scripts.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  render();
+}
+
+function render() {
+  const container = document.getElementsByClassName("container")[0];
+  while (container.hasChildNodes()) {
+    container.removeChild(container.firstChild);
+  }
+  myLibrary.forEach((book, index) => {
+    book.__proto__ = Object.create(new Book);
```
Comment (rvvergara) [inline assets/js/scripts.js]: What did we need this line for guys?

**Review by rvvergara — APPROVED** [review]
Hi @1ba1 and @gferrarocamus,

After an awesome Rails final project, great to see you in the JS world now! 

## Review notes:

- Did we use any popular style guide for our ESLint configurations? If not I suggest we do. We will catch most code that needs styling improvements. It will work wonders for your code :) 

- We have a very big `render` function. Let's break down its logic into smaller functions and then combine them into higher-order `render` function (function that calls other functions).

Other than these and the line of code where you did this `book.__proto__ = Object.create(new Book(...))` I think you guys satisfied the minimum requirements and have demonstrated JavaScript understanding. 

# :+1: 

You may move forward to your next adventure :)

File: assets/js/scripts.js (hunk 5 above)
Comment (1ba1) [reply assets/js/scripts.js]: Hi Ryan,
The problem is that when we retrieve data from the local storage we get objects literals, so we need to convert them in instances of Book, in order to make the toggle function work.

Comment (rvvergara) [reply assets/js/scripts.js]: I see. In that case the right way should have been:

```javascript
book.__proto__ = Book.prototype;
```

This converts your ordinary `book` object into an instance of `Book` :)

Comment (rvvergara) [reply assets/js/scripts.js]: In that case the better way should be:

```javascript
book.__proto__ = Book.prototype
```

Converting your ordinary `book` object into an instance of `Book`. Calling Book.new instantiates an entirely new object. :) This one doesn't

Comment (gferrarocamus) [reply assets/js/scripts.js]: Thanks, @rvvergara!
